src/spec-node/imageMetadata.ts

```diff
@@ -247,7 +281,20 @@ function collectOrUndefined<T, K extends keyof T>(entries: T[], property: K): No
 export function getDevcontainerMetadata(baseImageMetadata: SubstitutedConfig<ImageMetadataEntry[]>, devContainerConfig: SubstitutedConfig<DevContainerConfig>, featuresConfig: FeaturesConfig | undefined, omitPropertyOverride: string[] = []): SubstitutedConfig<ImageMetadataEntry[]> {
 	const effectivePickFeatureProperties = pickFeatureProperties.filter(property => !omitPropertyOverride.includes(property));
 
-	const raw = featuresConfig?.featureSets.map(featureSet => featureSet.features.map(feature => ({
+	// Variable substitution
+	// eslint-disable-next-line code-no-unused-expressions
+	featuresConfig?.featureSets.forEach(featureSet =>
+		featureSet.features.forEach(f => {
+			pickFeatureLifecycleHookProperties.forEach(hook => {
+				const buildPath = `/opt/build-features/${f.consecutiveId}`;
+				if (f[hook]) {
+					f[hook] = substituteFeatureRoot(f[hook], buildPath);
```

Comment:
After exploring this a bit, wouldn't it make more sense to continue to edit `raw`? The image metadata label stores the `raw`, 

```
[imageMetadataLabel]: JSON.stringify(getDevcontainerMetadata(imageBuildInfo.metadata, config, undefined).raw),
```

Which means that we lose the context of the where on disk the `featureRootFolder` is:

<img width="1823" alt="image" src="https://user-images.githubusercontent.com/23246594/218223366-9c430c6c-6e2d-431b-8572-3f69c28fa038.png">

${featureRootFolder} is not just the name of the Feature, but something we calculate (`consecutiveId`) - this was done to avoid name collisions.  

Do you still think it makes sense to not embed the path in the metadata.  That seems the most simple, but I could also store the path as a separate element in the metadata?

Reply:
@chrmarti - I've now made a change to not substitute the `raw` metadata, and instead cache an additional property `featureContainerDestFolder` that keeps track of the path within the container that the Feature is cached to.

I don't think there's currently a way around caching the path on the metadata, unless we rearchitect how the paths are crafted (without using the `consecutiveId`.  That change is a bit far-reaching and I think it would be better to leave that to further patch.  What do you think?

<img width="1299" alt="image" src="https://user-images.githubusercontent.com/23246594/218546512-1436409e-ee4d-41c1-8d44-77fa892e6e17.png">

Reply:
You're right. The updated version is even better. 👍